lib/multipart.js

```
@@ -441,8 +443,25 @@ Parser.handle = function(req, res, next, options) {
   });
 
   parser.on('data', function() {
-    if (this.writtenDisk > diskLimit || this.written > limit) {
-      this.emit('error', new Error('Overflow.'));
+    var err = null;
+    if (this.writtenDisk > diskLimit) {
+      err = new Error('Multipart disk overflow')
+      if (typeof options.handleMultipartDiskOverflow === "function") {
+        options.handleMultipartDiskOverflow(err, req, res, next);
+        err.hasCustomHandler = true;
+        throw err;
```

Just for my own knowledge- what's the reason why we want to throw the error here rather than emit an error event like below?

---

Emitting an error doesn't stop it from continuing to process the body, whereas throwing here will abort the entire process.